File: web/src/lib/currency.ts
export function centsToDisplay(cents: number): string {
    return (cents / 100).toFixed(2);
}

export function displayToCents(value: string): number {
    return Math.round(parseFloat(value.replace(",", ".")) * 100);
}

export function formatBRL(cents: number): string {
    return new Intl.NumberFormat("pt-BR", {
        style: "currency",
        currency: "BRL",
    }).format(cents / 100);
}

export function formatBRLCompact(cents: number): string {
    const value = cents / 100;
    if (value >= 1_000_000) return `R$ ${(value / 1_000_000).toFixed(1)}M`;
    if (value >= 1_000) return `R$ ${(value / 1_000).toFixed(1)}k`;
    return formatBRL(cents);
}

File: mobile/lib/core/utils/currency.dart
import 'package:intl/intl.dart';

/// Brazilian Real formatter. Displays centavos as "R$ X,XX".
final _brlFormat = NumberFormat.currency(
  locale: 'pt_BR',
  symbol: 'R\$',
  decimalDigits: 2,
);

/// Convert centavos [int] to a display [String], e.g. 150000 → "R$ 1.500,00".
String centsToDisplay(int cents) =>
    _brlFormat.format(cents / 100.0);

/// Convert a display string (from a masked text field) back to cents.
/// Strips "R$", ".", and replaces "," with ".".
int displayToCents(String display) {
  if (display.isEmpty) return 0;
  final cleaned = display
      .replaceAll(RegExp(r'[R$\s]'), '')
      .replaceAll('.', '')
      .replaceAll(',', '.');
  final parsed = double.tryParse(cleaned) ?? 0.0;
  return (parsed * 100).round();
}
